### PR TITLE
Fix #269672: Show selected notes on the Piano Keyboard

### PR DESCRIPTION
**Pull request by yukisaw**
https://musescore.org/en/node/269672

### CODE REVIEW

**Comment by lasconic** [comment]
Good start but

1. It doesn't work in the case of a range selection (Shift + Click)
2. It doesn't work in the case of multiple notehead selection (with Ctrl + Click)
3. What does "chordpart" means exactly ? it needs a better name
4. If I understand the use of `-n->pitch()` correctly, it's an ugly hack... it needs at least a comment, but it would be best to avoid  the hack altogether.

**Comment by yukisaw** [comment]
1. I don`t think this has to work on range selection.
2. Same as 1st. But probably I can add this like several active and inactive pitches at one time. That's not hard, but need a lot of "if" conditions.
3. Part of chord =) I will rename it when found a better name.
4. Need to change logic to avoid this.

And all these things will take much more than 15 minutes I spent on this code. So don`t know when the request will be done.

**Comment by anatoly-os** [comment]
@yukisaw so you did 80% of work for 20% of time. Since it is not must have functionality, it is better to spend more (free) time to finish it later, isn't it?

**Comment by yukisaw** [comment]
1. Skipped
2, 3 Resolved
4 Another hack.

File: mscore/pianotools.cpp [hunk 1]
```diff
@@ -437,5 +442,41 @@ bool HPiano::gestureEvent(QGestureEvent *event)
...
+                              pitches.insert(n == elements.front() ? n->pitch() : n->pitch() << 8);
+                        break;
+
+            default:    {
+                        if (elements.front()->type() != Ms::Element::Type::NOTE) break;
+                        QSet<Element*> selected;
+                        Chord* chord = static_cast<Note*>(elements.front())->chord();
+                        for (auto e : elements)
+                              {
+                              if (e->type() != Element::Type::NOTE || static_cast<Note*>(e)->chord() != chord)
```
Comment (Jojo-Schmitz) [inline mscore/pianotools.cpp]: in master we're using `isNote()` and `toNote()` instead, here and in other places of this PR

Comment (yukisaw) [reply mscore/pianotools.cpp]: There is no isNote() and toNote() methods at MuseScore 2.2

**Comment by yukisaw** [comment]
Last one. That`s totally stupid. Was very easy solution on 22 lines.

**Comment by anatoly-os** [comment]
In 3.0 piano code was fully reworked. So, it will be impossible to merge it to 3.0. And 3.0 has good methods like pressPitch() and releasePitch() without operations on ints. So, maybe implement it in 3.0?

File: mscore/pianotools.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply mscore/pianotools.cpp]: Ah, didn't notice this was o(only) for 2.2

**Comment by yukisaw** [comment]
You may make it by you own.

**Comment by anatoly-os** [comment]
@yukisaw I assume the last comment was addressed to me. I meant, implement it ONLY in 3.0 and avoid such unclear code in 2.2. I still don't understand the reason why we're implementing it.

**Comment by mattmcclinch** [comment]
I am sensing that there are still issues with this PR. So I would like to offer #3489 as an alternative.

**Comment by lasconic** [comment]
https://github.com/musescore/MuseScore/pull/3489 looks better.

**Pull request closed without merging**